[IMP] Sales: Updated product variant import language (#17830)
odoo/documentation: content/applications/sales/sales/products_prices/products/import.rst

The Odoo **Sales** app provides a template for importing products with categories and variants,
which can be opened and edited with any spreadsheet software (Microsoft Excel, OpenOffice, Google
Sheets, etc.) A properly formatted spreadsheet can be uploaded to an Odoo database. Within minutes,
all the products and their variants are live within Odoo and accessible in the product catalog.

Download template
=================

To import products with categories and variants, download the *Import Template for Products*. Once
downloaded, the template can be adjusted and customized, then uploaded into an Odoo database.

To download the template, navigate to :menuselection:`Sales app --> Products --> Products`. On the
:guilabel:`Products` page, click the :icon:`fa-cog` :guilabel:`(Action)` icon in the upper-left
corner. From the drop-down menu, select the :guilabel:`Import records` option.

> Selecting :guilabel:`Import records` opens a new page with a :guilabel:`Import Template for

Reviewer: ```suggestion
Selecting :guilabel:`Import records` opens a new page with an :guilabel:`Import Template for
```

Products` link to download the template. Once downloaded, open the spreadsheet file to customize it.

.. image:: import/import-template-products.png
   :alt: The Import records option selectable from the cog icon on the Products page in Odoo Sales.

Customize product import template
=================================

When modifying the product import template, keep in mind the following:

> - Columns that are unnecessary may be removed.

Reviewer: ```suggestion
- Unnecessary columns may be removed.
```

- Keeping the :guilabel:`Internal Reference` column is recommended.
- Having a unique identifier (e.g., `FURN_001`) for each product in the :guilabel:`Internal
  Reference` column can be helpful in many cases. This number can even be reused from previous
  software spreadsheets to ease the transition into Odoo. For example, when updating imported
  products, the same file can be imported several times without creating duplicates.
- Do **not** change the labels of columns that are meant to be imported. Otherwise, Odoo won't
  recognize them, requiring them to be configured manually on the import screen.
- New columns may be added to the template spreadsheet, if desired. However, newly added columns
  **must** exist in Odoo. If Odoo can't match a column's name with a field, it can be matched
  manually during the import process.

> When importing a completed template, Odoo displays a page that shows all the elements of the newly

Reviewer: ```suggestion
When importing a completed template, Odoo displays a page showing all the elements of the newly
```

configured product template spreadsheet separated by :guilabel:`File Column`, :guilabel:`Odoo
Field`, and :guilabel:`Comments`. To manually match a column name with a field in Odoo, click the
:guilabel:`Odoo Field` drop-down menu next to the :guilabel:`File Column` that needs a manual

> adjustment and select the appropriate field.

Reviewer 2: This section is helpful, especially the notes around Internal References and column matching! :D 

One thing I’m wondering, though: did Cedric mention whether these are common issues users specifically run into when importing product variants? Right now, some of this guidance seems broad enough that it could potentially belong on the more general import/export documentation rather than being product-variant specific.

If these recommendations are based on common pitfalls during product imports (duplicate products, broken field mapping, update issues, etc.), I think it would help to frame them more explicitly as "common import considerations" or "frequent product variant import issues to avoid." That would make it clearer to the reader why this guidance appears in this article specifically.

Otherwise, it may be worth considering whether some of these points should instead be referenced from the generic import/export docs in the essentials

Author: Ah, I actually did genericize this content some! My original conception for the product variants-specific content (I think it's in the task chatter) was that it would be its own thing, prior to putting it in the overall importation doc in Sales. I've updated the section heading for now, but I'll look into having a single source-of-truth in the Essentials section and linking to that instead.

fields are available on every tab of a product form. While these fields can be edited directly on
the product form, they can also be modified via a product import. Relation fields of this nature can
**only** be imported for products if the fields already exist in the database. For example, if a
user attempts to import a product with a *Product Type*, it must be one of the preconfigured product
types in the database (e.g., *Tracked Product*, *Untracked*, etc.).

To import information for a relation field on a product import template spreadsheet, add the name of
the field as a column name/title on the spreadsheet. Then add the desired relation field option on
the appropriate product line. When all desired relation field information has been entered, save the
spreadsheet and import it to the database. Go to the :menuselection:`Sales app --> Products -->
Products`. On the :guilabel:`Product` page click the :icon:`fa-cog` :guilabel:`(Action)` icon and
select :menuselection:`Import records --> Upload File`.

Once the spreadsheet with the newly-configured relation field information has been uploaded, click

> :guilabel:`Import`, and Odoo returns to the :guilabel:`Products` page. The new information on can be

Reviewer: ```suggestion
:guilabel:`Import`, and Odoo returns to the :guilabel:`Products` page. The new information can be
```

.. image:: import/attributes-and-values-spreadsheet.png
   :alt: An attributes and values spreadsheet template for imports.

- :guilabel:`Attribute`: The name of the attribute (e.g., `Size`).
- :guilabel:`Display Type`: The display type used in the product configurator. There are three
  display type options:

  - :guilabel:`Radio`: Values displayed as radio buttons.
  - :guilabel:`Selection`: Values displayed in a selection list.
  - :guilabel:`Color`: Values denoted as a color selection.

- :guilabel:`Variant Creation Mode`: The way variants are created when applied to a product. This

>   option **cannot** be changed once the attribute has been changed on a least one product. There are

Reviewer: ```suggestion
  option **cannot** be changed once the attribute has been changed on at least one product. There are
```

automatically created and added to the first corresponding attribute by name. Duplicated attribute
names may cause issues during this process.

.. example::
   The minimal amount of information to import product variants is as follows.

   The *Name* column defines the product template name and is used to group all the variants with
   the same name under a unique product template.

   The *Product Values* column defines the product template name and is used to group all the
   variants with the same name under a unique product template.

   =========  ========================
   Name       Product Values
   =========  ========================
   Odoo polo  Sleeves: Long, Size: S
   Odoo polo  Sleeves: Long, Size: M
   Odoo polo  Sleeves: Short, Size: S
   Odoo polo  Sleeves: Short, Size: M
   Odoo polo  Sleeves: Short, Size: L
   =========  ========================

.. important::
   Commas cannot be used in attribute and value names since they are used by the database to
   separate items from each other.

When product attributes and values are configured in the database, they can be used in product

> import spreadsheets to add more information and detail imported products.

Reviewer: ```suggestion
import spreadsheets to add more information and detail to imported products.
```

Attributes / Values`, and :guilabel:`Name` columns. These specific columns are **required** to
import products with specific variants properly.

.. image:: import/product-attribute-spreadsheet-import.png
   :alt: Product variants spreadsheet with product attributes and variants for import purposes.

- :guilabel:`Name`: The product's name.
- :guilabel:`Product Attributes / Attribute`: The name of the attribute.
- :guilabel:`Product Attributes / Values`: The values pertaining to the corresponding attribute.

.. tip::
   To import multiple values, separate them with a single comma (**not a comma followed by a
   space**) in the product import template spreadsheet (e.g., `furniture,couch,home`).

When the desired products and product variants have been entered and saved in the spreadsheet, they
can be imported into Odoo. To do that, navigate to :menuselection:`Sales app --> Products -->
Products`. On the :guilabel:`Products` page, click the :icon:`fa-cog` :guilabel:`(Action)` icon and
select :menuselection:`Import records --> Upload File`.

Once the spreadsheet with the newly-configured products and product variants has been uploaded,

> click :guilabel:`Import`. Odoo loads the :guilabel:`Products` page, where the newly-added products

Reviewer: ```suggestion
click :guilabel:`Import`. Odoo loads the :guilabel:`Products` page, where the newly added products
```
